src/pii-client.ts

```diff
@@ -0,0 +1,79 @@
+import PIIsdk, {
+    AgentType,
+    initAgent,
+    PIIEncryptionMethod,
+} from "@notabene/pii-sdk";
+import { TransactionArguments, TravelRuleOptions } from "./types";
+
+export class PIIEncryption {
+    private readonly config: TravelRuleOptions;
+    public toolset: PIIsdk;
+
+    constructor(config: TravelRuleOptions) {
+        this.config = config;
+        const requiredFields = [
+            "kmsSecretKey",
+            "baseURLPII",
+            "audiencePII",
+            "clientId",
+            "clientSecret",
+            "authURL",
+            "jsonDidKey",
+        ];
+        const missingFields = requiredFields.filter(
+            (field) => !(field in this.config)
+        );
+
+        if (missingFields.length > 0) {
+            throw new Error(
+                `Missing configuration fields: ${missingFields.join(", ")}`
+            );
+        }
+
+        this.toolset = new PIIsdk({
+            kmsSecretKey: config.kmsSecretKey,
+            piiURL: config.baseURLPII,
+            audience: config.audiencePII,
+            clientId: config.clientId,
+            clientSecret: config.clientSecret,
+            authURL: config.authURL,
+        });
+    }
+
+    async hybridEncode(transaction: TransactionArguments) {
+        const { travelRuleMessage } = transaction;
+        const pii = travelRuleMessage.pii || {
+            originator: travelRuleMessage.originator,
+            beneficiary: travelRuleMessage.beneficiary,
+        };
+        const { jsonBeneficiaryDidKey, jsonDidKey, kmsSecretKey } = this.config;
```

Comment:
`jsonBeneficiaryDidKey` should perhaps be renamed to `beneficiaryDidKey` because it's not a json formatted string but just a `did:key:...`

Reply:
Rename to beneficiaryDidKey hank you!